execute-sql-workitem/src/main/java/org/jbpm/process/workitem/executesql/ExecuteSqlWorkItemHandler.java

```diff
@@ -0,0 +1,130 @@
+import javax.sql.DataSource;
+
+import org.jbpm.process.workitem.core.AbstractLogOrThrowWorkItemHandler;
+import org.jbpm.process.workitem.core.util.RequiredParameterValidator;
+import org.jbpm.process.workitem.core.util.Wid;
+import org.jbpm.process.workitem.core.util.WidMavenDepends;
+import org.jbpm.process.workitem.core.util.WidParameter;
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.jbpm.process.workitem.core.util.service.WidAction;
+import org.jbpm.process.workitem.core.util.service.WidService;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Wid(widfile = "ExecuteSQLDefinitions.wid", name = "ExecuteSQL",
+        displayName = "ExecuteSQL",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.executesql.ExecuteSqlWorkItemHandler()",
+        documentation = "${artifactId}/index.html",
+        parameters = {
+                @WidParameter(name = "DataSource", required = true),
+                @WidParameter(name = "SQLStatement", required = true),
+                @WidParameter(name = "MaxResults"),
+                @WidParameter(name = "ColumnSeparator")
+        },
+        results = {
+                @WidResult(name = "Result")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+        },
+        serviceInfo = @WidService(category = "${name}", description = "${description}",
+                keywords = "database,fetch,sql,execute",
+                action = @WidAction(title = "Execute SQL statements")
+        ))
+public class ExecuteSqlWorkItemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(ExecuteSqlWorkItemHandler.class);
+    private static final String RESULT = "Result";
+    private static final int DEFAULT_MAX_RESULTS = 10;
+    private static final String DEFAULT_COLUMN_SEPARATOR = ",";
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+        try {
+            RequiredParameterValidator.validate(this.getClass(),
+                                                workItem);
+
+            Map<String, Object> results = new HashMap<>();
+            String dataSource = (String) workItem.getParameter("DataSource");
+            String sqlStatement = (String) workItem.getParameter("SQLStatement");
+            String maxResults = (String) workItem.getParameter("MaxResults");
+            String columnSeparator = (String) workItem.getParameter("ColumnSeparator");
+
+            Connection connection = null;
+            PreparedStatement statement = null;
+            ResultSet resultSet = null;
+
+            List<String> lines = new ArrayList<>();
+            try {
+                DataSource ds = InitialContext.doLookup(dataSource);
```

Comment:
move this up from the execute method into constructor so the data source look up will be done only once. If you think that single handler should use multiple data sources then I'd make it an option instead of default.